runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ComputationState.java

```diff
@@ -125,23 +126,28 @@ public boolean activateWork(ShardedKey shardedKey, Work work) {
   /**
    * Marks the work for the given shardedKey as complete. Schedules queued work for the key if any.
    */
-  public void completeWork(ShardedKey shardedKey, long workToken) {
-    Work nextWork;
+  public void completeWorkAndScheduleNextWork(ShardedKey shardedKey, long workToken) {
+    Optional<Work> nextWork;
     synchronized (activeWork) {
-      Queue<Work> queue = activeWork.get(shardedKey);
-      if (queue == null) {
+      Queue<Work> workQueue = activeWork.get(shardedKey);
```

Comment:
nullable annotation or optional

Reply:
made optional and moved into ActiveWorkState + tests